documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -50,6 +82,9 @@ The ``begin`` and ``end`` functions can take SYCL 2020 deduction tags and ``sycl
 to explicitly control which access mode should be applied to a particular buffer when submitting
 a SYCL kernel to a device:
 
+The return types of the ``begin`` and ``end`` functions are
+`indirectly device accessible <indirectly-device-accessible>`_.
```

Comment:
Does it also mean, that
`oneapi::dpl::is_indirectly_device_accessible<decltype(oneapi::dpl::begin(buf)))>`
is `std::true_type`?

Reply:
And the second question.
Does it also mean, that
1) `is_onedpl_indirectly_device_accessible_v` is true
`oneapi::dpl::begin(buf)` type ?

2) `oneapi::dpl::__ranges::__is_passed_directly_device_ready_v` is true for `oneapi::dpl::begin(buf)` type? if "yes", I am afraid that it is not quite correct with the implementation perspective.

Reply:
Yes, or they have the characteristics of `std::true_type`.  These types are not device copyable though, so they will not be "passed directly" to kernels, and will not operate differently than they ever have.

Reply:
`oneapi::dpl::begin(buf)` is not device copyable, this means `oneapi::dpl::__ranges::__is_passed_directly_device_ready_v` is `false`.  It contains a `sycl::buffer`.  The idea here was to have this trait / customization point provide the meaning: "representing data inherently accessible on the device", which is true of `oneapi::dpl::begin(buf)`.  

There was a discussion about this in the specification PR, this was at the suggestion of @akukanov .